XenModel/Actions/SR/SrAction.cs

```diff
@@ -35,139 +35,80 @@
 
 namespace XenAdmin.Actions
 {
-    public enum SrActionKind { SetAsDefault, Detach, Forget, Destroy, UnplugAndDestroyPBDs };
-
-    public class SrAction : AsyncAction
+    public interface ISrAction
     {
-        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+    }
 
-        private readonly SrActionKind kind;
 
-        public SrAction(SrActionKind kind, SR sr)
-            : base(sr.Connection, GetTitle(kind, sr))
+    public class SetSrAsDefaultAction : AsyncAction, ISrAction
+    {
+        public SetSrAsDefaultAction(SR sr)
+            : base(sr.Connection, string.Format(Messages.ACTION_SR_SETTING_DEFAULT, sr.Name(), Helpers.GetName(sr.Connection)))
         {
-            this.kind = kind;
             SR = sr;
-            Pool pool = Helpers.GetPoolOfOne(sr.Connection);
-            if (pool != null)
-                Pool = pool;
-            Host host = sr.GetStorageHost();
-            if (host != null)
-                Host = host;
-
-            switch (kind)
-            {
-                case SrActionKind.Detach:
-                    ApiMethodsToRoleCheck.Add("PBD.async_unplug");
-                    break;
-                case SrActionKind.Destroy:
-                    ApiMethodsToRoleCheck.Add("SR.async_destroy");
-                    break;
-                case SrActionKind.Forget:
-                    ApiMethodsToRoleCheck.Add("SR.async_forget");
-                    break;
-                case SrActionKind.SetAsDefault:
-                    ApiMethodsToRoleCheck.AddRange(
-                        "pool.set_crash_dump_SR",
-                        "pool.set_default_SR",
-                        "pool.set_suspend_image_SR");
-                    break;
-                case SrActionKind.UnplugAndDestroyPBDs:
-                    ApiMethodsToRoleCheck.AddRange("PBD.async_unplug", "SR.async_destroy");

```

Comment:
Here is the regression I mentioned at the descirption. It should be PBD.async_destroy, not SR. The practical result is the same because both calls are available to pool operators, but we should be using the correct call.